include/sentry.h

```diff
@@ -776,6 +777,32 @@ typedef sentry_value_t (*sentry_event_function_t)(
 SENTRY_API void sentry_options_set_before_send(
     sentry_options_t *opts, sentry_event_function_t func, void *data);
 
+/**
+ * Type of the `on_crash` callback.
+ *
+ * Does not work with crashpad on macOS.
+ * The callback passes a pointer to sentry_ucontext_s structure when exception
+ * handler is invoked. For breakpad on Linux this pointer is NULL.
+ *
+ * If the callback returns true outgoing crash report will be discarded.
```

Comment:
This might just be personal preference, but I think its also more consistent with similar callbacks:
Can you turn this logic around? So that `false` discards the event, and `true` will continue processing as normal?
This is also in line with how `before_send` behaves, as you return a `null` value to discard the event.

Reply:
I already inverted it to be consistent with crashpad and breakpad handler returns
But it makes sense, especially since it's not used for breakpad return value anymore
Though, maybe it also shouldn't be used for crashpad as well there?
https://github.com/getsentry/sentry-native/blob/e8011b66af01851175bb77a31773cbcb6f96f76e/src/backends/sentry_backend_crashpad.cpp#L188-L190

Reply:
I admit this is a bit confusing. Ideally the SDK users should not care about the underlying handler, and we can return an appropriate value to make the handler behave like it should.

Reply:
The problem is I'm not sure what is the desired behaviour.
The docs says 
`If the custom handler returns true, the signal is considered handled and the signal handler returns. Otherwise, the currently installed Crashpad signal handler is run.`
But the actual behaviour should depend on some formal requirements
I reverted the return to `false` for crashpad too. Let me know if you think it should depend on `on_crash` return

Reply:
> I reverted the return to false for crashpad too. Let me know if you think it should depend on on_crash return

Yes I think it should. Essentially we need to return `!should_handle`, as returning `true` from the crashpad hook tells it to not handle (aka send to sentry).

Reply:
alright, should be fine now